javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/SwitchEntryContext.java

```diff
@@ -78,19 +81,6 @@ public SymbolReference<? extends ResolvedValueDeclaration> solveSymbol(String na
             }
         }
 
-        // look for declaration in a pattern label for this entry
-        for (Expression e : wrappedNode.getLabels()) {
-            if (!e.isTypePatternExpr()) {
-                continue;
-            }
-
-            TypePatternExpr typePatternExpr = e.asTypePatternExpr();
-            if (typePatternExpr.getNameAsString().equals(name)) {
-                JavaParserPatternDeclaration decl = JavaParserSymbolDeclaration.patternVar(typePatternExpr, typeSolver);
-                return SymbolReference.solved(decl);
-            }
-        }
-
```

Comment:
I added this logic while adding support for switch pattern expressions. The possibility of having record patterns as well complicates this, however, so I moved the logic for finding declarations in patterns to `AbstractJavaParserContext` to avoid duplicating that. For that to work, I added the `typePatternExprsExposedToChild` method below which just finds all pattern exprs in the entry label and returns them (since they're all exposed to the children of this entry)